carp.deployment.core/src/commonMain/kotlin/dk/cachet/carp/deployment/application/DeploymentServiceHost.kt

```diff
@@ -71,8 +71,20 @@ class DeploymentServiceHost( private val repository: DeploymentRepository, priva
 
         val device = deployment.registrableDevices.firstOrNull { it.device.roleName == deviceRoleName }
             ?: throw IllegalArgumentException( "The specified device role name could not be found in the study deployment." )
-        deployment.registerDevice( device.device, registration )
 
+        // Early out when the device is already registered.
+        val priorRegistration = deployment.registeredDevices[ device.device ]
```

Comment:
If the `DeviceRegistration` would contain a notification token, Im not sure a client can guarantee to provide the same `DeviceRegistration`, on a redeployment.

Reply:
As discussed in the meeting, redeploying (deploying again after a previous successful deployment) will be considered a separate use case for which additional endpoints are required. But, we will allow registering devices with new `DeviceRegistration`'s as long as the deployment has not succeeded, to support erroneous deployments and making changed and attempting again.